os/arch/arm/src/amebad/amebad_watchdog_lowerhalf.c

```diff
@@ -0,0 +1,482 @@
+ * well-known watchdog_lowerhalf_s structure.
+ */
+struct amebad_wdg_lowerhalf_s {
+	FAR const struct watchdog_ops_s *ops;
+	xcpt_t handler;
+	uint32_t timeout_ms;
+	uint32_t timeout_left;		/* Record remaining timeout for PAUSE and RESUME */
+	uint32_t wdt_status;		/* 0: WDT_STOP; 1: WDT_START; 2: ON_PAUSE */
+	uint32_t start_tick;		/* Use to calculate timeleft before timeout */
+	uint32_t on_pause_duration;	/* Record the duration of watchdog timer on pause */
+	bool int_mode;				/* True for timeout interrupt mode on */
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+static int amebad_wdg_start(FAR struct watchdog_lowerhalf_s *lower);
+static int amebad_wdg_stop(FAR struct watchdog_lowerhalf_s *lower);
+static int amebad_wdg_keepalive(FAR struct watchdog_lowerhalf_s *lower);
+static int amebad_wdg_getstatus(FAR struct watchdog_lowerhalf_s *lower, FAR struct watchdog_status_s *status);
+static int amebad_wdg_settimeout(FAR struct watchdog_lowerhalf_s *lower, uint32_t timeout);
+static xcpt_t amebad_wdg_capture(FAR struct watchdog_lowerhalf_s *lower, xcpt_t handler);
+static int amebad_wdg_ioctl(FAR struct watchdog_lowerhalf_s *lower, int cmd, unsigned long arg);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+const struct watchdog_ops_s g_wdgops = {
+	.start = amebad_wdg_start,
+	.stop = amebad_wdg_stop,
+	.keepalive = amebad_wdg_keepalive,
+	.getstatus = amebad_wdg_getstatus,
+	.settimeout = amebad_wdg_settimeout,
+	.capture = amebad_wdg_capture,
+	.ioctl = amebad_wdg_ioctl,
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: amebad_wdg_start
+ *
+ * Description:
+ *   Start the watchdog timer, resetting the time to the current timeout,
+ *
+ * Input Parameters:
+ *   lower - A pointer the publicly visible representation of the "lower-half"
+ *           driver state structure.
+ *
+ * Returned Values:
+ *   Zero on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+static int amebad_wdg_start(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct amebad_wdg_lowerhalf_s *priv = (struct amebad_wdg_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+	if (!priv) {
+		return -ENODEV;
+	}
+	if (priv->wdt_status == WDT_STOP) {
+		/* Start watchdog timer */
+		watchdog_start();
+		priv->wdt_status = WDT_START;
+		priv->start_tick = SYSTIMER_TickGet();	// Record time right after wdt start
+		priv->on_pause_duration = 0;	// Reset on_pause_duration
+	}
+
+	return OK;
```

Comment:
Even if this timer is running, return value will be OK. Let's do like below
```
if (... == WDT_STOP) {
   start();
   return OK;
}

return -EBUSY;
```

Reply:
Sure, thanks!